common/nymsphinx/chunking/src/lib.rs

```diff
@@ -22,6 +27,118 @@ pub mod fragment;
 pub mod reconstruction;
 pub mod set;
 
+#[derive(Debug, Clone)]
+pub struct FragmentMixParams {
+    destination: PublicKey,
+    hops: u8,
+}
+
+impl FragmentMixParams {
+    pub fn destination(&self) -> &PublicKey {
+        &self.destination
+    }
+
+    pub fn hops(&self) -> u8 {
+        self.hops
+    }
+}
+
+#[derive(Debug, Clone, Serialize, ToSchema)]
+pub struct SentFragment {
+    header: FragmentHeader,
+    at: u64,
+    client_nonce: i32,
+    #[serde(skip)]
+    mixnet_params: FragmentMixParams,
+}
+
+impl SentFragment {
+    fn new(
+        header: FragmentHeader,
+        at: u64,
+        client_nonce: i32,
+        destination: PublicKey,
+        hops: u8,
+    ) -> Self {
+        let mixnet_params = FragmentMixParams { destination, hops };
+        SentFragment {
+            header,
+            at,
+            client_nonce,
+            mixnet_params,
+        }
+    }
+
+    pub fn header(&self) -> FragmentHeader {
+        self.header.clone()
+    }
+
+    pub fn at(&self) -> u64 {
+        self.at
+    }
+
+    pub fn client_nonce(&self) -> i32 {
+        self.client_nonce
+    }
+
+    pub fn seed(&self) -> i32 {
+        self.header().seed().wrapping_mul(self.client_nonce())
+    }
+
+    pub fn mixnet_params(&self) -> FragmentMixParams {
+        self.mixnet_params.clone()
+    }
+}
+
+#[derive(Debug, Clone, Serialize, ToSchema)]
+pub struct ReceivedFragment {
+    header: FragmentHeader,
+    at: u64,
+}
+
+impl ReceivedFragment {
+    fn new(header: FragmentHeader, at: u64) -> Self {
+        ReceivedFragment { header, at }
+    }
+
+    pub fn header(&self) -> FragmentHeader {
+        self.header.clone()
+    }
+
+    pub fn at(&self) -> u64 {
+        self.at
+    }
+}
+
+lazy_static::lazy_static! {
```

Comment:
nit: consider using `std::cell::OnceCell`

Reply:
yeah, with current changes to rust, you no longer need to use `lazy_static`, in 99.9% cases `OnceLock`/`LazyLock` (or the `Cell` variants if you don't need thread-safety) are sufficient

Reply:
Went with `LazyLock`, so that it is initialised on first access